WinCompatibilityPack/WinCompatibilityPack.psm1

```diff
@@ -0,0 +1,804 @@
+    "PnpDevice",
+    "PrintManagement",
+    "ProcessMitigations",
+    "Provisioning",
+    "PSScheduledJob",
+    "ScheduledTasks",
+    "SecureBoot",
+    "SmbShare",
+    "Storage",
+    "TrustedPlatformModule",
+    "VpnClient",
+    "Wdac",
+    "WindowsDeveloperLicense",
+    "WindowsErrorReporting",
+    "WindowsSearch",
+    "WindowsUpdate"
+)
+
+# Module-scope variable to hold the active compatibility session name
+$SessionName = $null
+
+# Specifies the default configuration to connect to when creating the compatibility session
+$DefaultConfigurationName = 'Microsoft.PowerShell'
+
+# Specifies the default name of the computer on which to create the compatibility session
+$DefaultComputerName = 'localhost'
+
+###########################################################################################
+<#
+.Synopsis
+   Initialize the connection to the compatibility session.
+.DESCRIPTION
+   Initialize the connection to the compatibility session. By default
+   the compatibility session will be created on the local host using the
+   'Microsoft.PowerShell' configuration. On subsequent calls, if a session
+   matching the current specification is found, it will be returned rather than
+   creating a new session. If a matching session is found, but can't be used, it
+   will be closed and a new session will be retrieved.
+
+   This command is called by the other commands in this module so
+   you will rarely call this command directly. 
+.EXAMPLE
+    Initialize-WinSession
+    Initialize the default compatibility session
+.EXAMPLE
+    Initialize-WinSession -ComputerName localhost -ConfigurationName Microsoft.PowerShell
+    Initialize the compatibility session with a specific computer name and configuration
+#>
+function Initialize-WinSession
+{
+    [CmdletBinding()]
+    [Alias("iwins")]
+    [OutputType("System.Management.Automation.Runspaces.PSSession")]
+    Param (
+
+        # If you don't want to use the default compatibility session, use
+        # this parameter to specify the name of the computer on which to create
+        # the compatibility session.
+        # (Defaults to 'localhost')
+        [Parameter(Mandatory=$false,Position=0)]
+        [String]
+        [Alias("Cn")]
+            $ComputerName = $script:DefaultComputerName,
+
+        # Specifies the configuration to connect to when creating the compatibility session
+        # (Defaults to 'Microsoft.PowerShell')
+        [Parameter()]
+        [String]
+            $ConfigurationName = $script:DefaultConfigurationName,
+
+        # The credential to use when connecting to the target machine/configuration
+        [Parameter()]
+        [PSCredential]
+            $Credential,
+
+        # If present, the specified session object will be returned
+        [Parameter()]
+        [Switch]
+            $PassThru,
+
+        # Instead of initializing the session, show what actions will be taken instead.
+        [Parameter()]
+        [Switch]
+            $WhatIf
```

Comment:
Is there some reason this is used instead of `SupportsShouldProcess`? Generally, using a manual whatif switch is frowned upon. I also don't see this being used anywhere in this function. It is better to not have it if it won't be used.

Reply:
Good catch. This was left over from an earlier refactoring of the code. WhatIf and Confirm are still used by Copy-WinModule where they are just passed to the underlying Copy-Item command. Still, I've changed that function to declare SupportsShouldProcess.